coderd/userauth_test.go

```diff
@@ -1673,27 +1682,186 @@ func TestUserForgotPassword(t *testing.T) {
 
 		anotherClient, anotherUser := coderdtest.CreateAnotherUser(t, client, user.OrganizationID)
 
-		err := anotherClient.RequestOneTimePasscode(ctx, codersdk.RequestOneTimePasscodeRequest{
+		// First try to login before changing our password. We expected this to error
+		// as we haven't change the password yet.
+		_, err := anotherClient.LoginWithPassword(ctx, codersdk.LoginWithPasswordRequest{
+			Email:    anotherUser.Email,
+			Password: newPassword,
+		})
+		require.Error(t, err)
```

Comment:
For the most resilience, be explicit about the error you're expecting to receive.
False-positives are your worst enemy in tests.

Reply:
I've made the error checking more explicit now!